Profiles, technical part

badboy opened the pull request:
Based on #165 (should merge cleanly afterwards, but I can also rebase it if necessary before this is merged)

[Rendered](https://badboy.github.io/firefox-data-docs-pcd/concepts/profile/profile_creation.html) (includes #165)
[My changes only](https://github.com/badboy/firefox-data-docs/compare/df51a0fd4772de2841f01de3d1506d9e9b0f35d2...badboy:pcd-creation-v2)

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+
+#### First use
+
+When Firefox is opened for the first time after a fresh install, without any prior Firefox profile on disk, it will create a new profile.
```
"When Firefox is opened for the first time after a fresh install, without any prior Firefox profile on disk"

"When Firefox is opened for the first time after a fresh install, without any prior Firefox profile on disk **visible to Firefox**"

- Firefox has some specific directories it checks for Profiles, right? Where it looks for / keeps profiles by default. Might be good to call that out.

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+
+On subsequent starts, Firefox tries to find the profile directory and load the profile data (bookmarks, addons, etc.).
+If not otherwise specified it uses the default (or last used) profile, finds the profile directory and starts loading files from there.
+
```
The above part seems out of place with this section. It's talking about how FF looks for profiles and decides which profile to use, while this bigger section is about how profiles are created. 

might be a good idea to break out a separate section for what a "Default" profile is and it's role

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+The directory can be changed by the user to another new or even an existing directory.
+If that directory does not exist, it is created.
+Additional files for bookmarks are placed inside this directory.
+
```
The wording is a bit confusing in this section. Would be good to simplify, like: 

"The profile manager allows the user to create a new profile from an existing directory (in which case any files will be included) or from scratch (in which case the directory will be created."

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+
+Firefox can be started on the command line with a path to a profile directory: `firefox --profile path/to/directory`.
+In that case the passed profile directory is, if not already existing, created. No files inside that directory are written at launch.
+
```
sorry, confused by the statement "No files inside that directory are written at launch."

SuYoungHong replied:
Also, one important distinction that we might want to mention is how with this method, this profile will **not persist** in the sense that the directory will remain but it won't be considered a 'profile' by Firefox (unless you run FF with this command to the same directory every time).

The directory you use as a 'profile' in this case will not be picked up by Profile Manager, whereas a profile created by default on first run of FF, or via Profile manager will be visible and available to profile manager afterwards

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+
+A profile can be deleted by simply removing the profile directory.
+We will never know about a deletion. We simply won't see that profile turning up in Telemetry data anymore.
+
```
Might want to include how, just so it's explicitly clear. Profile deletion via: 
* Profile manager
* Deleting the profile directory
* Deleting Firefox does NOT delete profile directory

SuYoungHong commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+### During Profile Creation
+
+When a profile is created explicitly (default profile on first launch, new profile from Profile Manager or the `--createprofile` on the command line), the profile directory is created and a `times.json` containing a timestamp of the current time is stored inside that profile directory[^1].
+It is read at later times when the profile creation date is used.
```
this category of profile creation (created explicitly) seems super useful, would be great if we could define it in the profile creation section above:

created explicity
* default profile on first launch
* new profile from Profile Manager
* --createprofile command

not created explicitly (?)
* --profile

SuYoungHong replied:
it would help understanding the flowchart a lot

georgf commented on high_level/profile/profile_creation.md:
```diff
@@ -0,0 +1,112 @@
+
+These different cases are described in more detail in the following sections.
+
+### Profile Creation
```
Firefox Developer edition [automatically sets up a separate profile](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Developer_Edition) for you, so you can easily run it in parallel with Firefox Release.
Can we call this out as well?

georgf commented:
Chris, can you take a look at the technical correctness?
This is not 100% complete yet (aiming to publish rather than wait for 100%), some sections will be filled in later.

badboy commented:
I addressed @SuYoungHong's and @georgf's initial feedback. Turning it into explicit/implicit profile creation makes it much easier to follow (and refer back) to the ways it's done.

badboy commented on concepts/profile/profile_creation.md:
```diff
@@ -0,0 +1,138 @@
+## Profile Creation Date
+
+The *profile creation date* is the assumed date of initial profile creation.
+However it proofed to be not reliable for all cases.
```
Thanks. English is hard.

badboy commented on concepts/profile/profilehistory.md:
```diff
@@ -0,0 +1,96 @@
+
+![](images/profile-history/example-midhistory.png)
+
+In this example, the profile simply appears in our data mid-history, with presumably the 25th subsession in it's history. It's previous history is a mystery. 
```
I mark this as a follow-up, so we can properly phrase that.

badboy replied:
#171

badboy commented on concepts/profile/profilehistory.md:
```diff
@@ -0,0 +1,96 @@
+
+![](images/profile-history/example-restart-2.png) 
+
+Here, we see the profile start their history normally, but then they begin a new, totally unconnected branch with a starting ping that is **not** the same as the original starting ping (different `subsession_id`s). 
```
The included image shows that the profile_reset_date is None.
I'll file a follow-up to keep track of that.

badboy replied:
https://github.com/mozilla/firefox-data-docs/issues/172

badboy commented on concepts/profile/realworldusage.md:
```diff
@@ -0,0 +1,135 @@
+* 6: Firefox installation
+	- The installed Firefox program on a given disk image. 
+	- Since Firefox is usually installed in a system directory, the single installation of Firefox will be shared by all the OS user profiles in the disk image. 
+	- Stub installers are instrumented with pings to report new install counts, however, full binaries are not. 
```
https://github.com/mozilla/firefox-data-docs/issues/173

badboy commented on concepts/profile/realworldusage.md:
```diff
@@ -0,0 +1,135 @@
+As with the Multi-Profile User and Shared Computer case, even though there could be multiple Firefox profiles in this use case, there will only be one download and install event.  
+
+#### Migrations
+
```
https://github.com/mozilla/firefox-data-docs/issues/174

badboy commented:
Addressed all comments regarding profile creation here, the other one are included in #165.

This is now only blocked on getting #165 merged and then needs a rebase before merge.

badboy commented:
@SuYoungHong and @chutten were okay with the new phrasing of managed/unmanaged. I'm now gonna land this.
